machinist/mserver/controller.go

```diff
@@ -146,24 +147,29 @@ func (en *Controller) addNodeToDns(name string, ips []net.IP, tags []string) {
 	}
 }
 
-func (en *Controller) ServeAllRecords() {
+func (en *Controller) ServeAllRecords(killChannel chan struct{}) {
 	for {
-		ns := en.Nodes()
-		for _, d := range en.dnsServer.Domains {
-			dnsName := dns.CanonicalName(fmt.Sprintf("%s.%s", "_all", d))
-			var rs []dns.RR
-			for _, v := range ns {
-				for _, i := range v.Ips {
-					rr, err := dns.NewRR(fmt.Sprintf("%s %s %s", dnsName, "A", i.String()))
-					if err != nil {
-						en.Log.Errorf("err: %v", err)
+		select {
+		case <-time.After(en.allRecordsRefreshRate):
```

Comment:
This works, but consider instead using a time.Ticker for two reasons:

* the interval time is even if the loop body takes a long time to run relative to the refresh rate (currently the actual interval is `allRecordsRefreshRate` + the execution time of the loop body)
* bigger reason: the ticker can be (probably wrapped with some sort of interface, and then) passed in so that this can be tested without relying on timing. In prod, pass an actual time.Ticker, but in tests, pass a ticker that requires a call to emit each tick, so you can tick, assert state, tick, assert new state, etc.